src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1044,9 +1044,15 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                 transactionStartedByUs = true;
             }
 
-            this.expressionMap.queryEntity = true;
+            if (!this.expressionMap.selectDistinct)
+                this.expressionMap.queryEntity = true;
```

Comment:
can you please explain these lines? Its not obvious why we need to enable this flag

Reply:
Previously this flag was always set as enabled in getRawAndEntities. When this flag is enabled the id of the selected table is added to the query. This means the selection is no longer distinct. Does this flag need to be enabled by default?

Reply:
If the Id could be omitted with select would it remove the need for these lines?